app/services/doc_auth/lexis_nexis/doc_pii_reader.rb

```diff
@@ -32,23 +32,23 @@ def read_pii(true_id_product)
           month: pii.delete(:dob_month),
           day: pii.delete(:dob_day),
         )
-        pii[:dob] = dob if dob
+        pii[:dob] = dob
 
         exp_date = parse_date(
           year: pii.delete(:state_id_expiration_year),
           month: pii.delete(:state_id_expiration_month),
           day: pii.delete(:state_id_expiration_day),
         )
-        pii[:state_id_expiration] = exp_date if exp_date
+        pii[:state_id_expiration] = exp_date
 
         issued_date = parse_date(
           year: pii.delete(:state_id_issued_year),
           month: pii.delete(:state_id_issued_month),
           day: pii.delete(:state_id_issued_day),
         )
```

Comment:
this one of those "as long as you're here...." kinda ideas, not needed for this PR but maybe we could do as a follow-up.

 I don't like the idea of adding all the birthday fields to the main hash and then deleting them. What if we broke out the date keys from `PII_INCLUDES` and had something like...

```
PII_INCLUDES_DATES = {
  state_id_expiration: ['Fields_ExpirationDate_Year', 'Fields_ExpirationDate_Month', 'Fields_xpirationDate_Day'] #  this is NOT a typo
  },
}

pii = PII_INCLUDES.map do |true_id_key, idp_key|
  [idp_key, true_id_product[:IDAUTH_FIELD_DATA][true_id_key]]
end.to_h

# would be cool to do this in a way that doesn't mutate either, but w/e for now
pii[:state_id_type] = DocAuth::Response::ID_TYPE_SLUGS[pii[:state_id_type]]

dates = PII_INCLUDES_DATES.transform_values do |year_month_day_keys|
  Date.new(*true_id_product[:IDAUTH_FIELD_DATA].values_at(*year_month_day_keys))
end

Pii::StateId.new(**pii.merge(dates))
```

Reply:
Yeah, this could definitely work better but I think I may move it to a follow-up. A wrinkle I found taking a quick swing at it is we do not trust TrueID to give us valid dates in all cases:

https://github.com/18F/identity-idp/blob/862abf7dcd0b7c161ec443ecb48cce45e314b606/app/services/doc_auth/lexis_nexis/doc_pii_reader.rb#L78-L86

Reply:
oh right 😭 yeah we could swap in the `parse_date` method in there but definitely good follow-up candidate

Reply:
I took a shot at cleaning up this method here: https://github.com/18F/identity-idp/pull/10515. I didn't go with this approach there because I found it easier to just get rid of this map and do everything inline in the constructor. PTAL!